gateways/istio-egress/templates/deployment.yaml

```diff
@@ -47,7 +47,7 @@ spec:
 {{- else }}
           image: "{{ .Values.global.hub }}/{{ .Values.global.proxy_init.image | default "proxy_init" }}:{{ .Values.global.tag }}"
 {{- end }}
-          imagePullPolicy: IfNotPresent
+          imagePullPolicy: {{ .Values.global.imagePullPolicy | default "Always" }}
```

Comment:
After thinking about @costin's argument, I believe the best approach would be to change the default in `values.yaml` and then sed in the release artifacts process.  This simplifies this PR significantly.  Im not confinced the default "Always" is working the way it should..

This does lead to the problem that what we release is not what we tag, so I'd like @fpesce 's insight into how to solve that problem.  Until he chimes in, i'd atleast like to block this PR for backport until that is sorted out.  (perhaps post 1.2).  If you remove the backports flag, we can merge earlier.  I think just simply making the image pull policy configurable would get the job done, along with changing the default pull policy.

Reply:
a different (imo better) approach is to change the default in values.yaml for the CI, and then sed post CI (as a final PR prior to release).

Thoughts?

Reply:
I think it is a good practice to use | default ".." - it is easier to read and understand the template, if you need to go back to values.yaml for each word to find it's real value it's a pain. 

For the release: with the new installer we support Kustomize and kubect apply -k - which implies that in the release branch where we use v1.2.x we need to set IfNotInstalled and the real version when we generate the files for kustomize.

Master branch - Always is needed since we use a floating label ( latest-daily )

Reply:
well I can't see I agree.  If a change in defaults occurs, it must be changed all over the manifests, vs just one place.  Still I don't care strongly enough to block this prl.

Cheers
-steve